Content.Server/Administration/Managers/BanManager.cs

```diff
@@ -186,16 +207,32 @@ public async void CreateServerBan(NetUserId? target, string? targetUsername, Net
     // Removing it will clutter the note list. Please also make sure that department bans are applied to roles with the same DateTimeOffset.
     public async void CreateRoleBan(NetUserId? target, string? targetUsername, NetUserId? banningAdmin, (IPAddress, int)? addressRange, ImmutableArray<byte>? hwid, string role, uint? minutes, NoteSeverity severity, string reason, DateTimeOffset timeOfBan)
     {
-        if (!_prototypeManager.TryIndex(role, out JobPrototype? _))
+        string? prefix = null;
+        var antagAllSelection = Loc.GetString("ban-panel-role-selection-antag-all-option");
+
+        if (_prototypeManager.TryIndex<JobPrototype>(role, out _))
+        {
+            prefix = JobPrefix;
+        }
+
+        else if (_prototypeManager.TryIndex<AntagPrototype>(role, out _) || role == antagAllSelection)
+        {
+            prefix = AntagPrefix;
+        }
+
+        if (prefix != null)
+        {
+            role = string.Concat(prefix, role);
+        }
+        else
```

Comment:
At this point just make the caller handle it because if a job + antag overlap it's just going to cook bans entirely.